README.md

```diff
@@ -19,9 +35,155 @@ PipelineRL tackles the classic trade-off between **inference throughput** (large
+                        " After each guess I will tell you if your answer is higher or lower than the target number."
+        }
+    ]
+    time_start = time.time()
+    llm_calls = []
+    guess_history = []
+    reward = 0
+    success = 0
+    error = 0
+    for i in range(13):
+        messages = initial_messages.copy()
+        if i > 0:
+            last_message = f"Your {i} previous guesses:"
+            for guess in guess_history:
+                relation = "lower" if guess < problem["answer"] else "higher"
+                last_message += f"\n{guess}, which is {relation} than the target number."
+            else:
+                last_message += "\n<wrong output>"
+            messages.append({
+                "role": "user",
+                "content": last_message
+            })
+        llm_call = await llm_async_generate(llm, Prompt(messages=messages), session)
+        llm_calls.append(llm_call)
+
+        output_text = llm_call.output.content or ""
+        answer = re.search("<answer>(\d+)</answer>", output_text)
+        if answer:
+            answer = int(answer.group(1))
+            if answer == problem["answer"]:
+                reward = 2 - i / 10
+                success = 1
+                break
+            else:
+                guess_history.append(answer)                            
+        else:
+            # bonus for using the correct output format in the first turns
+            reward = -2 + i / 10
+            error = 1
+            break
+    latency = time.time() - time_start        
+
+    # TrainingText contains the prompt and output tokens, reward, and the log probs of the output tokens necessary for RL training.
+    training_texts = [make_training_text(llm, llm_call) for llm_call in llm_calls]
+    for text in training_texts:
+        text.reward = reward
+
+    metrics = BaseMetrics(
+        reward=reward,
+        success=success,
+        no_error=not error,
+        no_answer=error,
+    )
+
+    return RolloutResult(
+        training_texts=training_texts,
+        metrics=metrics,
+        latency=latency,
+        dataset_name=problem["dataset"],
+    )
+    
+````
+
+Finally you need to create a Hydra config file that points to the rollout function and the dataset loader. Additional hyper-parameters such as model path, learning rate, etc. can also be modified. For example, `guessing.yaml`:
+
+````yaml
+defaults:
+    - base
+    - _self_
+
+actor:
+    rollout_policy: pipelinerl.domains.guessing.generate_guessing_rollout
+environment: null
+dataset_loader: pipelinerl.domains.guessing.load_problems
+train_dataset_names:
+    - train
+test_dataset_names:
+    - test
+````
+
+You can now launch the training with the following command:
+
+```bash
+python -m pipelinerl.launch config_name=guessing output_dir=results/guessing
+```
+
+After a few minutes, the actor will first be evaluated on the test dataset, and then it will start collecting training rollouts. 
```

Comment:
Not clear what would happen during this "few minutes" before the evaluation starts.